src/asm.rs

```diff
@@ -164,3 +164,48 @@ pub fn ttat(addr: *mut u32) -> u32 {
 pub unsafe fn bx_ns(addr: u32) {
     call_asm!(__bxns(addr: u32));
 }
+
+/// Semihosting syscall.
+///
+/// This method is used by cortex-m-semihosting to provide semihosting syscalls.
+#[inline]
+pub unsafe fn sh_syscall(nr: u32, arg: u32) -> u32 {
```

Comment:
Since this is public API, maybe a clearer name would be appropriate?

```suggestion
pub unsafe fn semihosting_syscall(nr: u32, arg: u32) -> u32 {
```